controllers/clustercache/cluster_accessor_client.go

```diff
@@ -99,17 +100,27 @@ func (ca *clusterAccessor) createConnection(ctx context.Context) (*createConnect
 		}
 	}
 
+	log.V(6).Info("Creating uncached client (finalized)")
```

Comment:
WDYT about instead doing the following in l.101 (inside the if):
```go
		log.V(6).Info(fmt.Sprintf("Creating uncached client with updated REST config with host %q", restConfig.Host))
		uncachedClient, err = createUncachedClient(ca.config.Scheme, restConfig, httpClient, mapper)
		if err != nil {
			return nil, errors.Wrapf(err, "error creating uncached client (using in-cluster config)")
		}
```

\+ moving newClientWithTimeout inside the createUncachedClient func?

Reply:
Makes sense. We use the same uncachedClient var throughout and avoids double-creation.